tensorboard/plugins/pr_curve/tf_pr_curve_dashboard/tf-pr-curve-card.ts

```diff
@@ -0,0 +1,160 @@
+    // we actually want the value itself to be a function.
+    _xComponentsCreationMethod: {
+      type: Object,
+      value: () => (() => {
+        const scale = new Plottable.Scales.Linear();
+        return {
+          scale: scale,
+          axis: new Plottable.Axes.Numeric(scale, 'bottom'),
+          accessor: (d: PrCurvePoint) => d.recall,
+        };
+      }),
+      readOnly: true,
+    }
+  },
+  observers: [
+      "reload(run, tag)",
+      "_runsChanged(_attached, runs.*)",
+      "_setChartData(_runToData, stepCap)",
+  ],
+  _computeRunColor(run) {
+    return this._colorScaleFunction(run);
+  },
+  _serializedRunToPrCurveData(runToPrCurveData) {
+    return JSON.stringify(runToPrCurveData);
+  },
+  attached() {
+    // Defer reloading until after we're attached, because that ensures that
+    // the requestManager has been set from above. (Polymer is tricky
+    // sometimes)
+    this._attached = true;
+    this.reload();
+  },
+  reload() {
+    if (!this._attached) {
+      return;
+    }
+    this._canceller.cancelAll();
+    const router = getRouter();
+    let url = router.pluginRoute("pr_curve", "/pr_curves");
+    url += url.indexOf('?') > -1 ? '&' : '?';
+    url += `tag=${this.tag}&runs=${this.runs.join(',')}`;
+    const updateData = this._canceller.cancellable(result => {
+      if (result.cancelled) {
+        return;
+      }
+
+      // This is a mapping between run and data.
+      const runToData = result.value;
+      this.set('_runToData', runToData);
+      this.fire('data-updated', {
+        'runToData': runToData,
+      });
+    });
+    this.requestManager.request(url).then(updateData);
+  },
+  _setChartData(runToData, stepCap) {
+    if (!runToData || !_.isNumber(stepCap)) {
+      return;
+    }
+
+    _.forOwn(runToData, (entries, run) => {
+      if (!entries || !entries.length) {
+        return;
+      }
+
+      let seriesData: PrCurvePoint[] = [];
+      let entriesIndex = entries.length - 1;
+      while (entriesIndex > 0) {
+        if (entries[entriesIndex].step <= stepCap) {
+          // This is the rightmost entry closest (or equal to) the step cap.
+          break;
+        }
+        entriesIndex--;
+      }
+      const entry = entries[entriesIndex];
+      this.fire('run-step-updated', {
+        'run': run,
+        'step': entry.step,
+      });
+
+      // Reverse the values so they are plotted in order, which allows for
+      // tool tips.
+      entry.precision.reverse();
+      entry.recall.reverse();
+
+      for (let i = 0; i < entry.precision.length; i++) {
+        seriesData.push({
+          scalar: entry.precision[i],
+          recall: entry.recall[i],
+        });
+      }
+      this.$$('vz-line-chart').setSeriesData(run, seriesData);
+    });
+  },
+  _runsChanged(attached, runsUpdateRecord) {
+    if (!attached) {
+      return;
+    }
+    this.$$('vz-line-chart').setVisibleSeries(this.runs);
+    this.reload();
+  },
+  _computeLineChartStyle(loading) {
```

Comment:
I'd prefer to reflect `loading` to an attribute and use a CSS rule to handle this instead. The fewer dumb Polymer functions we have, and the less our code [looks like this](https://github.com/tensorflow/tensorboard/blob/5d0b9766ea6e7a879178a2e2b27c81bcdb117cf6/tensorboard/plugins/image/tf_image_dashboard/tf-image-loader.html#L243-L263), the better.